File: DailyAffirmation/Models/DayEntry.swift
import Foundation
import SwiftData

@Model
final class DayEntry: Identifiable {
    var id: UUID
    var date: Date
    var thought: String
    var affirmation: String?
    var createdAt: Date
    var updatedAt: Date

    init(
        id: UUID = UUID(),
        date: Date,
        thought: String = "",
        affirmation: String? = nil,
        createdAt: Date = Date(),
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.date = date.startOfCalendarDay
        self.thought = thought
        self.affirmation = affirmation
        self.createdAt = createdAt
        self.updatedAt = updatedAt
    }
}

File: web/src/models/dayEntry.ts
import { calendarDayKey } from '../utils/dates'

export interface DayEntry {
  id: string
  dateKey: string
  thought: string
  affirmation: string | null
  createdAt: string
  updatedAt: string
}

export function createDayEntry(date: Date, thought = ''): DayEntry {
  const now = new Date().toISOString()
  return {
    id: crypto.randomUUID(),
    dateKey: calendarDayKey(date),
    thought,
    affirmation: null,
    createdAt: now,
    updatedAt: now,
  }
}
